.github/workflows/bin/spack-reviewers.py

```
@@ -34,8 +38,11 @@ def main():
             package = match.group(1)
 
             # add maintainers from packages here
-            pkg_maintainers = spack.repo.PATH.get_pkg_class(package).maintainers
-            maintainers.update(pkg_maintainers)
+            try:
+                pkg_maintainers = spack.repo.PATH.get_pkg_class(package).maintainers
+                maintainers.update(pkg_maintainers)
+            except spack.repo.UnknownPackageError:
```

```suggestion
            except Exception:
```

---

Why should we backoff to the exception catchall here? Are there other possible errors in Spack here that we're likely to hit? My preference would be to not catch all so that we're made aware of changes to the core codebase instead of having this check always pass.

---

I think printing something about the error here would be good. But I agree wtih @alecbcs that catching the specific expected error mode is better here, if this fails for some other unexpected reason it should make some noise.

---

Then catch RepoError or SpackError, which can be raised when code loading fails. I'm suggesting a generic exception because it's private api and lately a few exceptions were moved between modules.

---

If it does break, it will only break on a bump PR and it can be fixed there as well. The point is we won't suppress unexpected errors, so if spack updates and creates a new edge case it will be more apparent when it is hit so the check can be appropriately updated.